IncludeToolbox/Commands/TryAndErrorRemoval_Project.cs

```diff
@@ -112,19 +103,39 @@ private bool ProcessNextFile()
                     continue;
 
                 impl.PerformTryAndErrorRemoval(document, settings);
-                ++fileIdx;
                 return true;
             }
-
-            projectItems = null;
             return false;
         }
 
+        static void RecursiveFindFilesInProject(ProjectItems items, ref Queue<ProjectItem> projectFiles)
+        {
+            var e = items.GetEnumerator();
+            while (e.MoveNext())
+            {
+                var item = e.Current;
+                if (item == null)
+                    continue;
+                var projectItem = item as ProjectItem;
+                if (projectItem == null)
+                    continue;
+                if (projectItem.FileCount > 1)
+                    RecursiveFindFilesInProject(projectItem.ProjectItems, ref projectFiles);
+                else if (projectItem.FileCount == 1)
+                {
+                    projectFiles.Enqueue(projectItem);
+                }
+            }
+        }
+
         private void PerformTryAndErrorRemoval(Project project)
         {
             projectItems = project.ProjectItems;
 
-            if (projectItems.Count > 2)
+            projectFiles.Clear();
+            RecursiveFindFilesInProject(projectItems, ref projectFiles);
+
+            if (projectFiles.Count > 10)
```

Comment:
Ahh, I should have changed it back.
BUT, the list is not only .cpp files, so you might get a popup saying it will take a long time, and actually have zero files that can be compiled.
Prefiltered, and lower number?

2 sounds good atm. :)